src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -5,97 +5,713 @@
+                        .build();
+        return ceToSend;
+    }
+
+    /**
+     * Creates a continuous delivery repository modified event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param name
+     * @param owner
+     * @param url
+     * @param view
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * repository modified event extensions
+     */
+
+    public static CloudEvent createRepositoryModifiedEvent(
+            final String eventType, final String id,
+            final URI source, final String name,
+            final String owner, final URI url,
+            final URI view, final String data) {
+        CloudEvent ceToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source",source)
+                        .withExtension("name", name)
+                        .withExtension("owner",owner)
+                        .withExtension("url", url)
+                        .withExtension("view", view)
+                        .build();
         return ceToSend;
     }
 
+    /**
+     * Creates a continuous delivery repository deleted event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param name
+     * @param owner
+     * @param url
+     * @param view
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * repository deleted event extensions
+     */
+
+    public static CloudEvent createRepositoryDeletedEvent(
+            final String eventType, final String id,
+            final URI source, final String name,
+            final String owner, final URI url,
+            final URI view, final String data) {
+        CloudEvent ceToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source",source)
+                        .withExtension("name", name)
+                        .withExtension("owner",owner)
+                        .withExtension("url", url)
+                        .withExtension("view", view)
+                        .build();
+        return ceToSend;
+    }
+
+    /**
+     * Creates a continuous delivery branch created event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * branch created event extensions
+     */
+    public static CloudEvent createBranchCreatedEvent(
+            final String eventType, final String id,
+            final URI source, final String repository,

```

Comment:
`repository` should be an object of Repository, defined at https://github.com/cdevents/spec/blob/v0.1.0/source-code-version-control.md